### PR TITLE
fix confusion in service_controller

### LINKED ISSUE #51009: Confused code and comment in serviceController
According to code snippet https://github.com/kubernetes/kubernetes/blob/e633a1604f00908a1dcc898b206c3404db4d82ed/pkg/controller/service/service_controller.go#L263-L290

The `createLoadBalancerIfNeeded` handles the **ServiceUpdate** events, it first check out whether or not current service need a LoadBalancer, and if not, it ensures corresponding LoadBalacner deleted.  

However, another branch of the `if` statement is a little bit confusing. It says _The load balancer doesn't exist yet, so create it_, and then it calls `s.createLoadBalancer`. I think at this time, we're not sure whether or not the load balancer already exists (since we didn't call `GetLoadBalancer`), so the proper behavior would be *ensure load balancer exists* (actually `s.createLoadBalancer` just calls `EnsureLoadBalancer` indeed), so the comment (L-283) and the function name (L-285 `createLoadBalancer`) isn't very appropriate, it's confusing right?

### PR DESCRIPTION
**Pull request by dastanza**
**What this PR does / why we need it**:

Fix code and comment confusion in `service_controller`.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #51009 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/controller/service/service_controller.go [hunk 1]
```diff
@@ -280,13 +280,13 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 
 		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
 
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, v1.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-		newState, err = s.createLoadBalancer(service)
+		// Ensure existence of the load balancer
```
Comment (yastij) [inline pkg/controller/service/service_controller.go]: This comment doesn't add much value now I think.

**Comment by yastij** [comment]
this lgtm generally. @MrHohn to you for review/approval.

File: pkg/controller/service/service_controller.go (hunk 1 above)
Comment (MrHohn) [inline pkg/controller/service/service_controller.go]: Probably just remove this comment? Also other than ensures LB exist, ensureLoadBalancer() could also ensure LB updated :)

**Comment by dunjut** [comment]
@MrHohn @yastij I've removed the comment line, please check out again. :)

**Comment by MrHohn** [comment]
Thanks, this LGTM.

/retest
/lgtm

**Comment by yastij** [comment]
@dunjut since review is done, please squash your commit for approval

**Comment by dunjut** [comment]
@yastij It's my first time contributing to kubernetes, to be confirmed, do you mean `git reset` and `git commit` and then `git push -f`? is there any commit message convention to be respected?

**Review by yastij — APPROVED** [review]
LGTM

**Comment by yastij** [comment]
@dunjut - you should wait for k8s approvers for lgtm/approve

**Pull request merged**